src/MuxManager.cpp

```diff
@@ -507,4 +514,65 @@ void MuxManager::generateServerMac(uint16_t serverId, std::array<uint8_t, ETHER_
     }
 }
 
+// ---> updateWarmRestartReconciliationCount(int increment);
+//
+// update warm restart reconciliation count
+//
+void MuxManager::updateWarmRestartReconciliationCount(int increment)
+{
+    MUXLOGDEBUG(increment);
+
+    boost::asio::io_service &ioService = mStrand.context();
+
+    ioService.post(mStrand.wrap(boost::bind(
+        &MuxManager::handleUpdateReconciliationCount,
+        this,
+        increment
+    )));
+}
+
+// ---> handleUpdateReconciliationCount(int increment);
+//
+// handler of updating reconciliation port count
+// 
+void MuxManager::handleUpdateReconciliationCount(int increment)
+{
+    MUXLOGDEBUG(mPortReconciliationCount);
+
+    mPortReconciliationCount += increment;
+
+    if(mPortReconciliationCount == 0) {
+        mReconciliationTimer.cancel();
+    } 
+}
+
+// ---> startWarmRestartReconciliationTimer
+//
+// start warm restart reconciliation timer
+// 
+void MuxManager::startWarmRestartReconciliationTimer(uint32_t timeout)
+{
+    mReconciliationTimer.expires_from_now(boost::posix_time::seconds(
+        timeout == 0? mMuxConfig.getMuxReconciliationTimeout_sec():timeout
+    ));
+    mReconciliationTimer.async_wait(mStrand.wrap(boost::bind(
+        &MuxManager::handleWarmRestartReconciliationTimeout,
+        this,
+        boost::asio::placeholders::error
+    )));
+}
+
+// ---> handleWarmRestartReconciliationTimeout
+// 
+// handle warm restart reconciliationTimeout
+//
+void MuxManager::handleWarmRestartReconciliationTimeout(const boost::system::error_code errorCode)
+{
+    if (errorCode == boost::system::errc::success) {
+        MUXLOGWARNING("Reconciliation timed out after warm restart, set service to reconciled now.");
+    }
+
+    mDbInterfacePtr->setWarmStartStateReconciled();
```

Comment:
In the case of timeout, some ports might be set as `auto` and some set as `manual`. Is that correct? If yes, is that expected? Should the timeout logic set the pending ports also as `auto`?